src/jvm/test/org/jetbrains/kotlinx/lincheck_test/gpmc/IntrinsicCandidateMethodTest.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.gpmc
+
+import org.jetbrains.kotlinx.lincheck.ExperimentalModelCheckingAPI
+import org.jetbrains.kotlinx.lincheck.runConcurrentTest
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+import kotlin.concurrent.thread
+
+/**
+ * Here we aim to test the problem with jit optimizing out calls
+ * to local object manager. Which used to cause invalid thread switch attempt.
+ * The issues used to easily reproduce on old jdks (e.g. 8, 11).
+ *
+ * See [Invalid thread switch attempt](https://github.com/JetBrains/lincheck/issues/576).
+ */
+@OptIn(ExperimentalModelCheckingAPI::class)
+class IntrinsicCandidateMethodTest {
+
+    @Test
+    fun testArraysCopyOf() {
+        runConcurrentTest(10000) {
+            val threads = mutableListOf<Thread>()
+            val l = CountDownLatch(1)
```

Comment:
Is `CountDownLatch` necessary for the test to fail (without intrinsic methods support)?
If so, please add a comment about it.

Reply:
I have rewritten test using atomic integer to simplify it. Added a comment which explains that the operation inside each thread does not matter, but with some it is easier to trigger jit

Reply:
On the second thought: the same test with `AtomicInteger` sometimes passes, and it is harder to reproduce the issue with it. So I will stick to `CountDownLatch` and add a comment that with primitves issue exists but sometimes does not reproduce